src/core/retryer.ts

```diff
@@ -163,7 +166,10 @@ export class Retryer<TData = unknown, TError = unknown> {
           // Do we need to retry the request?
           const retry = config.retry ?? 3
           const retryDelay = config.retryDelay ?? defaultRetryDelay
-          const delay = functionalUpdate(retryDelay, this.failureCount) || 0
+          const delay =
+            typeof retryDelay === 'function'
+              ? retryDelay(this.failureCount, error)
+              : retryDelay
```

Comment:
I am not sure why `|| 0` was needed here before. On type level, the function was only allowed to return a `number`, and we also use the `defaultRetryDelay` in case `retryDelay` was `undefined`